morpheus/_lib/doca/src/doca_source.cpp

```diff
@@ -68,158 +68,290 @@ std::optional<uint32_t> ip_to_int(std::string const& ip_address)
-        auto src_port_out_d      = rmm::device_uvector<uint16_t>(MAX_PKT_RECEIVE, rmm::cuda_stream_default);
-        auto dst_port_out_d      = rmm::device_uvector<uint16_t>(MAX_PKT_RECEIVE, rmm::cuda_stream_default);
-        auto tcp_flags_out_d     = rmm::device_uvector<int32_t>(MAX_PKT_RECEIVE, rmm::cuda_stream_default);
-        auto ether_type_out_d    = rmm::device_uvector<int32_t>(MAX_PKT_RECEIVE, rmm::cuda_stream_default);
-        auto next_proto_id_out_d = rmm::device_uvector<int32_t>(MAX_PKT_RECEIVE, rmm::cuda_stream_default);
-        auto timestamp_out_d     = rmm::device_uvector<uint32_t>(MAX_PKT_RECEIVE, rmm::cuda_stream_default);
-        auto exit_condition = std::make_unique<morpheus::doca::DocaMem<uint32_t>>(m_context, 1, DOCA_GPU_MEM_GPU_CPU);
-
-        auto fixed_width_inputs_table_view = cudf::table_view(std::vector<cudf::column_view>{
-            cudf::column_view(cudf::device_span<const int64_t>(src_mac_out_d)),
-            cudf::column_view(cudf::device_span<const int64_t>(dst_mac_out_d)),
-            cudf::column_view(cudf::device_span<const int64_t>(src_ip_out_d)),
-            cudf::column_view(cudf::device_span<const int64_t>(dst_ip_out_d)),
-            cudf::column_view(cudf::device_span<const uint16_t>(src_port_out_d)),
-            cudf::column_view(cudf::device_span<const uint16_t>(dst_port_out_d)),
-            cudf::column_view(cudf::device_span<const int32_t>(tcp_flags_out_d)),
-            cudf::column_view(cudf::device_span<const int32_t>(ether_type_out_d)),
-            cudf::column_view(cudf::device_span<const int32_t>(next_proto_id_out_d)),
-            cudf::column_view(cudf::device_span<const uint32_t>(timestamp_out_d)),
+        struct packets_info *pkt_ptr;
+        int sem_idx = 0;
+        cudaStream_t rstream = nullptr;
+        cudaStream_t pstream = nullptr;
+        cudf::table_view *fixed_width_inputs_table_view[MAX_SEM_X_QUEUE];
+
+        std::vector<rmm::device_uvector<char>> payload_buffer_d;
+        std::vector<rmm::device_uvector<int32_t>> payload_sizes_d;
+        std::vector<rmm::device_uvector<int64_t>> src_mac_out_d;
+        std::vector<rmm::device_uvector<int64_t>> dst_mac_out_d;
+        std::vector<rmm::device_uvector<int64_t>> src_ip_out_d;
+        std::vector<rmm::device_uvector<int64_t>> dst_ip_out_d;
+        std::vector<rmm::device_uvector<uint16_t>> src_port_out_d;
+        std::vector<rmm::device_uvector<uint16_t>> dst_port_out_d;
+        std::vector<rmm::device_uvector<int32_t>> tcp_flags_out_d;
+        std::vector<rmm::device_uvector<int32_t>> ether_type_out_d;
+        std::vector<rmm::device_uvector<int32_t>> next_proto_id_out_d;
+        std::vector<rmm::device_uvector<uint32_t>> timestamp_out_d;
+
+        int thread_idx = mrc::runnable::Context::get_runtime_context().rank();
+
+        if (thread_idx >= MAX_QUEUE) {
+            LOG(ERROR) << "Thread ID " << thread_idx << " bigger than MAX_QUEUE " << MAX_QUEUE << std::endl;
```

Comment:
We should never get here unless we have a bug further up in the code, so let's do
```suggestion
            MORPHEUS_FAIL(MORPHEUS_CONCAT_STR("Thread ID " << thread_idx << " bigger than MAX_QUEUE " << MAX_QUEUE));
```